pkg/liveshare/port_forwarder.go

```diff
@@ -112,20 +102,17 @@ func (fwd *PortForwarder) Forward(ctx context.Context, conn io.ReadWriteCloser)
 }
 
 // Connects to and pings a given address to ensure that the server is shared and the port is forwarded.
-func WaitForPortConnection(ctx context.Context, address string) error {
-	waitCtx, cancel := context.WithTimeout(ctx, connectionTimeout)
+func WaitForPortConnection(ctx context.Context, address string, timeout time.Duration) error {
```

Comment:
No need for the errgroup, defers in loops are bad news because they won't run until the end of the function, not the end of the loop

```go
func WaitForPortConnection(ctx context.Context, address string) {
	for {
		select {
		case <-ctx.Done();
			return ctx.Err()
		default:
			err := connectedToAddr(address)
			if err != nil {
				continue
			}
			
			return nil // success
		}
	}
}

func connectedToAddr(address string) error {
	// Verify that the port can be connected to
	conn, err := net.Dial("tcp", address)
	if err != nil {
		return err
	}
	defer conn.Close()
	
	// Send a ping and make sure it succeed
	_, err = conn.Write([]byte("ping"))
	if err != nil {
		return err
	}
	
	return nil
}
```

Reply:
Yeah, have the caller do it. Look at my suggestion here: https://github.com/cli/cli/pull/6358#discussion_r991584729 - prob needs a smaller timeout value but that's it

Reply:
Ah nevermind, we can use `context.WithTimeout` in the caller instead!